app/utils/pager.go

```diff
@@ -12,47 +12,45 @@ type Pager struct {
 	Next      int64
 	IsPrev    bool
 	IsNext    bool
+	IsValid bool
 }
 
 func NewPager(page, size, total int64) *Pager {
-	if page < 1 {
-		page = 1
-	}
-	p := new(Pager)
-	p.Current = page
-	p.Size = size
-	p.Total = total
-	p.Pages = total / size
-	if total%size > 0 {
-		p.Pages += 1
-	}
-	p.PageSlice = make([]int64, p.Pages)
+	pager := new(Pager)
+	pager.Current = page
+	pager.Size = size
+	pager.Total = total
+	pager.Pages = total / size
```

Comment:
@samdfonseca Yeah, `size = 0` will result in a 500 error, thanks for pointing it out.

This function looks messy, I will tidy it up and add test case when `size = 0`.